crux_core/src/core/mod.rs

```diff
@@ -72,7 +72,8 @@ where
     where
         Op: Operation,
     {
-        let _ = request.resolve(result);
+        let effects = request.resolve(result);
+        debug_assert!(effects.is_ok());
```

Comment:
Small thing (i assume I'm just misunderstanding something) - but since this is named effects it kind of suggests there are effects inside the result.  Which we're then just ignoring and doing nothing with?  Bit odd looking

Reply:
Err... I'm being dumb. Effects come out of the next call, this returns a `Result<(), ResolveError>` 🤦🏻‍♂️